### ISSUE TITLE
dfhack.designations.markPlant doesn't allocate a priority

### ISSUE THREAD

**Issue by PatrikLundell** [DFHack/dfhack#1905]
The function designates a plant, but it doesn't allocate a priority to the task. The "manual" method I've used previously didn't either, so I'm no worse off, but it would be an improvement if the operation could allocate a priority, as it seems designations without a priority doesn't end up anywhere in the priority scheme, i.e. it seems these priority less designations can end up before priority 1 as well as after priority 7.

**Comment by lethosor** [comment]
Yep, this is an oversight on my part. There is some [existing code in MapCache](https://github.com/DFHack/dfhack/blob/22a0b62e4317eb8ee046014186e627ea63ad6142/library/modules/MapCache.cpp#L304-L312) that deals with priorities. That could be used here, but setting up and tearing down a MapCache might be expensive, from looking at the constructor briefly. Maybe these priority functions could be relocated to the Designations module?

**Comment by PatrikLundell** [comment]
I'm not sure this is actually a bug, as it looks more like an improvement to me.

It certainly looks like overkill to go via MapCache, although I have no idea if it actually matters, i.e. if the extra work would take enough time to be noticeable for the player, but it would make sense to provide the functionality without the overhead.

I've noticed two things with the MapCache code after a brief look at it:
- The preceding operation, PriorityAt() returns "false" when "block" isn't defined, which is ugly for an int32_t returning function.
- There's no functionality to remove an "empty" priority event if setting the priority to 0 would leave it without any priorities, and it even creates an event if there wasn't one when the "null" priority is set. I don't know if the former ought to be done, but the latter ought to be prevented, which I think is done by switching "true" to "false" in "auto event = getPriorityEvent(block, true);", followed by "if (!event) return false;" (or possibly returning "true", depending on whether you want a noop to report success or failure, with "true" preserving the current function).

**Comment by lethosor** [comment]
Going with the precedent set by #481, I'm inclined to call it a bug, even if it's minor in this case.

* That `false` was simply a mistake, maybe copied from setPriorityAt or another function. It should be `0` instead, which isn't a valid priority as far as DF is concerned.
* I believe this matches DF's behavior, but I could be wrong. If there isn't a designation, it doesn't matter whether the priority is reset to 0 or not, and I don't think DF bothers to clean up unnecessary block events.

In any case, I'd be in favor of addressing this with new functionality in the designations module.
